src/session-pool.ts

```diff
@@ -758,6 +793,8 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
         });
       });
     }
+
+    span.addEvent(`Requested for ${nRequested} sessions returned ${nReturned}`);
```

Comment:
what is the need of creating `nRequested` and  `nReturned` , we can dircetly use `amount` and `sessions.length` respectively

Reply:
Sadly amount gets reduced on each iteration, so we can't use it, hence why we capture it upfront.